ui/app/templates/components/job-version.hbs

```diff
@@ -10,7 +10,7 @@
   </span>
   <div class="pull-right">
     {{#unless this.isCurrent}}
-      {{#if (can "run job")}}
+      {{#if (can "run job" namespace=this.jobNamespace)}}
```

Comment:
So this is the crux of everything. 

Instead of thinking about this in terms of "can the user run a job for the namespace set on the page that this version is rendered on", we can think of it as "can the user run a job for the namespace this version belongs to".

To do that we have to do the following:

1. Get the job the version belongs to
2. Get the namespace the version belongs to
3. Use that namespace here

Luckily, all of these relationships are already modeled in Ember data. [Job versions have a `belongsTo` for their job](https://github.com/hashicorp/nomad/blob/main/ui/app/models/job-version.js#L5) and [Jobs have a `belongsTo` for their namespace])https://github.com/hashicorp/nomad/blob/main/ui/app/models/job.js#L121).

So we can instead write this line as:

```hbs
{{{#if (can "run job" namespace=this.version.job.namespace)}}
```

And we're done. No prop-drilling, no query params, no routes, no controllers.

Reply:
How did you make this sound so simple?